versioned/persist/mongodb/src/main/java/org/projectnessie/versioned/persist/mongodb/MongoDatabaseAdapterConfig.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (C) 2020 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.versioned.persist.mongodb;
+
+import javax.annotation.Nullable;
+import org.immutables.value.Value;
+import org.projectnessie.versioned.persist.adapter.DatabaseAdapterConfig;
+
+@Value.Immutable(lazyhash = true)
+public interface MongoDatabaseAdapterConfig extends DatabaseAdapterConfig {
+
+  @Nullable
+  String getConnectionString();
+
```

Comment:
Missing `MongoDatabaseClient withConnectionString(String connectionString)` +  `MongoDatabaseClient withDatabaseName(String databaseName)`

Reply:
Why are they required?

Reply:
Just completeness - every *DatabaseAdapterConfig sub-type has those.

Reply:
Added "with" methods in a1c226a4c4965e21b3c581d638a8fc302d8a4061